mumax3c/scripts/driver.py

```diff
@@ -48,8 +50,31 @@ def driver_script(driver, system, compute=None, **kwargs):
             mx3 += f"gammaLL = {gamma0/mm.consts.mu0}\n"
             mx3 += "doprecess = true\n"
 
+        zh_li_terms = system.dynamics.get(type=mm.ZhangLi)
+        if zh_li_terms:
+            u = (
+                zh_li_terms[0].u
+                if isinstance(zh_li_terms[0].u, df.Field)
+                else df.Field(
+                    mesh=system.m.mesh,
+                    dim=3,
+                    value=(1.0, 0.0, 0.0),
+                    norm=zh_li_terms[0].u,
+                )
+            )
+
+            j = -np.multiply(
+                u
+                * (mm.consts.e / (mm.consts.e * mm.consts.hbar / (2.0 * mm.consts.me))),
+                system.m.norm,
+            )
+            j.write("j.ovf")
+            mx3 += f"Xi = {system.dynamics.get(type=mm.ZhangLi)[0].beta}\n"
```

Comment:
```suggestion
        if system.dynamics.get(type=mm.ZhangLi):
            (zh_li_term,) = system.dynamics.get(type=mm.ZhangLi)
            u = (
                zh_li_term.u
                if isinstance(zh_li_term.u, df.Field)
                else df.Field(
                    mesh=system.m.mesh,
                    dim=3,
                    value=(1.0, 0.0, 0.0),
                    norm=zh_li_term.u,
                )
            )

            j = -np.multiply(
                u
                * (mm.consts.e / (mm.consts.e * mm.consts.hbar / (2.0 * mm.consts.me))),
                system.m.norm,
            )
            j.write("j.ovf")
            mx3 += f"Xi = {zh_li_term.beta}\n"
```

The unpacking (second line) implicitly ensures that we only have one ZhangLi term in the dynamics equation.